storage/filestore.go

```diff
@@ -0,0 +1,205 @@
+package storage
+
+import (
+	"fmt"
+	"github.com/docker/notary"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+// NewFilesystemStore creates a new store in a directory tree
+func NewFilesystemStore(baseDir, subDir, extension string) (*FilesystemStore, error) {
+	baseDir = path.Join(baseDir, subDir)
+
+	return NewFileStore(baseDir, extension, notary.PrivKeyPerms)
+}
+
+// NewFileStore creates a fully configurable file store
+func NewFileStore(baseDir, fileExt string, perms os.FileMode) (*FilesystemStore, error) {
+	baseDir = filepath.Clean(baseDir)
+	if err := createDirectory(baseDir, perms); err != nil {
```

Comment:
It ensures we have the permissions to create it earlier in the workflow. We can fail fast and early if we fail here. Rather than possibly putting the user through creating a key and selecting a password, before eventually failing on `Set`